### PR TITLE
Add support FullColor (24bit) for Windows

### PR DESCRIPTION
**Pull request by prifio**
I add api for support 24bit color and 8bit color(256 color). Change struct Attribute to class Attribute and add class FullAttibute : Attribute. Add FullColor class (just r-g-b) and static method to convert 24bit color &larr;&rarr; 8bit color (Linux-like) &larr;&rarr; 4bit color (Old-windows-like). 
Change WindowsDriver implementation for support 24bit color.
But i can't implement checking windows version (old verson don't support 24bit color). I tried to use registry, but .Net Core don't support it. (see SetSupportFullColor() in WindowsConsole)
Change background of Box10x in example.
image

### CODE REVIEW

**Comment by migueldeicaza** [comment]
Thanks for the contribution!   I am currently on vacation, with limited time to review, but I think this looks good.

I will take some code, but I suspect that this might force me to finally revisit the color allocation code, and come up with something better.

**Comment by prifio** [comment]
I rewrote some part of code. I removed FullAttribute and back to struct Attibute and expanded Attribute (it's strange, but I more time didn't see easy way). Refactored code, restored cursor position after ConsoleWrite. And I tried implement TrueColor support detection. `System.Runtime.InteropServices.RuntimeInformation.OSDescription` Work fine (unlike `System.Environment.OSVersion`), but CI server can't build bkz he can't find this property.

File: Terminal.Gui/Drivers/ConsoleDriver.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using Mono.Terminal;
 using NStack;
 using Unix.Terminal;
+using System.Drawing;
```
Comment (migueldeicaza) [inline Terminal.Gui/Drivers/ConsoleDriver.cs]: System.Drawing will be problematic, I would rather avoid bringing another assembly dependency.

**Comment by migueldeicaza** [comment]
This looks great.   I will check after this 0.16 bug fix releease

File: Terminal.Gui/Drivers/ConsoleDriver.cs (hunk 1 above)
Comment (prifio) [reply Terminal.Gui/Drivers/ConsoleDriver.cs]: Sorry. I forgot remove it. Now, System.Drawing not used

**Comment by tig** [comment]
Is this a PR that we should take? Is this still valid?

**Comment by tig** [comment]
@migueldeicaza - please comment on this PR. Should we pursue it or abandon?

**Comment by AndreR** [comment]
Pretty sad, that this was abandoned!

**Comment by tig** [comment]
I agree. It would be awesome if someone took it on. The tracking issue is here: https://github.com/migueldeicaza/gui.cs/issues/48

**Comment by AndreR** [comment]
@tig What do you think? How complicated would it be to re-implement this pull request into the current state of development?

**Pull request closed without merging**